frontend/app/src/components/account-management/LoginForm.vue

```diff
@@ -361,26 +365,55 @@ const primaryProgress: ComputedRef<Progress | null> = computed(() => {
 
 const { tc } = useI18n();
 
-const customBackendRules = [
-  (v: string) => !!v || tc('login.custom_backend.validation.non_empty'),
-  (v: string) =>
-    (v &&
-      v.length < 300 &&
-      /^https?:\/\/(www\.)?[-a-zA-Z0-9@:%._+~#=]{1,256}(\.[a-zA-Z0-9()]{1,6})?\b([-a-zA-Z0-9()@:%_+.~#?&/=]*)$/.test(
-        v
-      )) ||
-    tc('login.custom_backend.validation.url')
-];
-
-const usernameRules = [
-  (v: string) => !!v || tc('login.validation.non_empty_username'),
-  (v: string) =>
-    (v && /^[0-9a-zA-Z_.-]+$/.test(v)) || tc('login.validation.valid_username')
-];
-
-const passwordRules = [
-  (v: string) => !!v || tc('login.validation.non_empty_password')
-];
+const rules = {
+  username: {
+    required: helpers.withMessage(
+      tc('login.validation.non_empty_username'),
+      required
+    ),
+    isValidUsername: helpers.withMessage(
+      tc('login.validation.valid_username'),
+      (v: string): boolean => !!(v && /^[0-9a-zA-Z_.-]+$/.test(v))
+    )
+  },
+  password: {
+    required: helpers.withMessage(
+      tc('login.validation.non_empty_password'),
+      required
+    )
+  },
+  customBackendUrl: {
+    required: helpers.withMessage(
+      tc('login.custom_backend.validation.non_empty'),
+      requiredIf(customBackendDisplay)
+    ),
+    isValidUrl: helpers.withMessage(
+      tc('login.custom_backend.validation.url'),
+      (v: string): boolean =>
+        !get(customBackendDisplay) ||
+        (v.length < 300 &&
```

Comment:
do you think we can mix and match a few rules and maybe use the URL validator https://vuelidate-next.netlify.app/validators.html#url instead of this regex?

Reply:
tried, that validator doesn't accept port